action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -242,6 +245,7 @@ func (stateDB *StateDBAdapter) SubBalance(evmAddr common.Address, amount *big.In
 
 // AddBalance adds balance to account
 func (stateDB *StateDBAdapter) AddBalance(evmAddr common.Address, amount *big.Int) {
+	stateDB.lastAddBalanceAmount.SetUint64(0)
```

Comment:
what's the purpose?

Reply:
to clear the amount for each AddBalance(), consider the following:
1. AddBalance(4000)
2. AddBalance(0)
3. Inside Suicide, it calls AddBalance(4000) to beneficiary but failed, if the amount is not cleared, it will mistakenly think the amount match and issue a tx log with amount=4000, but the 4000 is in step 1 and is mistaken